File: backend/src/main/java/com/djuancito/reposteria/entidad/Contacto.java
package com.djuancito.reposteria.entidad;

import java.time.LocalDateTime;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "Contacto")
public class Contacto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "contactoId")
    private Integer contactoId;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "telefono")
    private String telefono;

    @Column(name = "email")
    private String email;

    @Column(name = "mensaje")
    private String mensaje;

    @Column(name = "fecha")
    private LocalDateTime fecha;

    @Enumerated(EnumType.STRING)
    @Column(name = "estado")
    private EstadoContacto estado;
}

enum EstadoContacto { pendiente, respondido }

File: frontend/src/app/modelos/contacto.ts

export interface Contacto {
  contactoId: number;
  nombre: string;
  telefono: string;
  email: string;
  mensaje: string;
  fecha: string; 
  estado: 'pendiente' | 'respondido';
  motivo: 'cotizacion' | 'soporte' | 'general';
  respuesta?: string;    
  fechaRespuesta?: string;
}